Code/Framework/AzCore/AzCore/Math/Frustum.cpp

```diff
@@ -236,4 +237,20 @@ namespace AZ
 
         return viewFrustumAttributes;
     }
+
+    bool Frustum::GetCorners(CornerVertexArray& corners) const
+    {
+        using namespace ShapeIntersection;
```

Comment:
I did it this way more because it's also returning a bool to let you know if the operation succeeded, and it mirrors the way `IntersectThreePlanes()` works. Technically people could put anything in a frustum's planes to make it very not-frustum-like, so I wanted a way to let users know when it didn't work.

On the other hand we could just leave it up to the user - you pass in bad data, you might get bad data out. Honestly I'm fine doing it either way.

Reply:
I love `optional` for this (if the return can fail) but I know it's not for everyone, just a suggestion so feel free to leave as-is if for consistency this is better to use